src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlySpanND.cs

```diff
@@ -0,0 +1,468 @@
+            _linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            Debug.Assert(_linearLength >= 0);
+
+            _reference = ref reference;
+
+            _lengths = lengths.ToArray();
+            _strides = strides.ToArray();
+            _isPinned = isPinned;
+        }
+
+        /// <summary>
+        /// Returns a reference to specified element of the ReadOnlySpanND.
+        /// </summary>
+        /// <param name="indices"></param>
+        /// <returns></returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when index less than 0 or index greater than or equal to Length
+        /// </exception>
+        public ref readonly T this[params ReadOnlySpan<nint> indices]
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if (indices.Length != Rank)
+                    ThrowHelper.ThrowIndexOutOfRangeException();
+
+                var index = SpanNDHelpers.GetIndex(indices, Strides, Shape);
+                return ref Unsafe.Add(ref _reference, index /* force zero-extension */);
+            }
+        }
+
+        /// <summary>
+        /// The number of items in the span.
+        /// </summary>
+        // REVIEW: Calling this just Length for now based on our discussions. Can rename if we desire.
+        public nint Length => _linearLength;
+
+        /// <summary>
+        /// Gets a value indicating whether this <see cref="SpanND{T}"/> is empty.
+        /// </summary>
+        /// <value><see langword="true"/> if this span is empty; otherwise, <see langword="false"/>.</value>
+        public bool IsEmpty => _linearLength == 0;
+
+        /// <summary>
+        /// Gets a value indicating whether this <see cref="SpanND{T}"/> is pinned.
+        /// </summary>
+        /// <value><see langword="true"/> if this span is pinned; otherwise, <see langword="false"/>.</value>
+        public bool IsPinned => _isPinned;
+
+        // REIVEW: Calling this Shape for now as Lengths didn't seem to be the most liked option based on our discussion. Can rename if we desire.
+        /// <summary>
+        /// Gets the length of each dimension in this <see cref="SpanND{T}"/>.
+        /// </summary>
+        public ReadOnlySpan<nint> Shape => _lengths;
+
+        /// <summary>
+        /// Gets the rank, aka the number of dimensions, of this <see cref="SpanND{T}"/>.
+        /// </summary>
+        public int Rank => Shape.Length;
+
+        /// <summary>
+        /// Gets the strides of this <see cref="SpanND{T}"/>
+        /// </summary>
+        public ReadOnlySpan<nint> Strides => _strides;
+
+        /// <summary>
+        /// Returns false if left and right point at the same memory and have the same length.  Note that
+        /// this does *not* check to see if the *contents* are equal.
+        /// </summary>
+        public static bool operator !=(ReadOnlySpanND<T> left, ReadOnlySpanND<T> right) => !(left == right);
```

Comment:
nit: I would expect `operator ==` and `operator !=` to live directly adjacent to eachother in the code.

Reply:
Fixed.